utils/dd_to_csv.py

```diff
@@ -124,8 +123,8 @@ def parse_parameter_values_from_file(
 
 
 def save_data_with_headers(
-    param_data_dict: Dict[str, Union[pd.DataFrame, List[str]]],
-    headers_data: Dict[str, List[str]],
+    param_data_dict: dict[str, pd.DataFrame | list[str]],
```

Comment:
What python versions are you planning to support?
This was done with the `py11` setting (to match CI), but I'll revert and re-run for `py39` if that's the minimum target.  (I think the `|` type hint only works from 3.10 onwards)

Reply:
We haven't had that discussion actually... How about `py12`?

Reply:
Works for me, type hints are a lot cleaner in 3.12 for instance.
I just thought someone mentioned a use-case for running with an older version is all.

Reply:
I think we use the match-case statement somewhere in `transforms.py` which is >= 3.11. I feel perhaps we should try to keep the min version as low as possible, just to make it easier for users? Not sure how much of a pain it is to install a specific Python version in Windows (in linux I use `pyenv`)

Reply:
I'd say it is easy! :-) But I am also fine with sticking to 3.11.

Reply:
Yeah I use pyenv on windows also, works very well.
Less savvy users can just download and run the official installers etc.
Ok, so we'll target >=3.11, cool.